### DISCUSSION TITLE
Slow down and freezing on large vector PDF

### DISCUSSION THREAD

**Discussion by Monarchtest** [sumatrapdfreader/sumatrapdf#3841] [General]
Hello!
I downloaded the new version 3.5.2, something is getting worse and worse. The thing is that I have a lot of PDF files with a lot of vector graphics (tens of thousands of objects). Sumatra PDF version 3.2 opens such files instantly and does not slow down when enlarging / navigating through such a PDF drawing. But new versions, 3.4.2 and 3.5.2, slow down to the point of freezing.
Example: https://cloud.mail.ru/public/RZ3y/cfUdUQFnD

**Comment by GitHubRulesOK** [comment]
Yes has been noted and raised as an issue https://github.com/sumatrapdfreader/sumatrapdf/issues/3842, but SumatraPDF is a balancing act that seesaws between often differing resource requirements.
https://github.com/sumatrapdfreader/sumatrapdf/issues/3795

Using your sample on this slow machine it takes Acrobat in Edge about 60 seconds to render
SumatraPDf takes a tad more than 6 seconds
```
DisplayModel::BuildPagesInfo took 0.01 ms
Slow rendering: 6112.41 ms, page: 1 in 'C:\Users\lez\Downloads\Apps\PDF\SumatraPDF\issues\situation _empty.pdf'
```


![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/394015ee-d2aa-4a7f-9a31-7743fd4c6770)

**Comment by Monarchtest** [comment]
I don't compare with other programs. I compare versions specifically of Sumatra PDF: on version 3.2 everything loads very quickly, and on later versions it loads very slowly. For myself, I will only use version 3.2 for now. I hope that in future versions this problem will be corrected, for now I would call it a regression.
